### PR TITLE
Add trustchain

### PR DESCRIPTION
**Pull request by TimeEngineer**

### CODE REVIEW

File: oxidation/crates/parsec_api_types/src/error.rs [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    Crypto { exc: CryptoError },
+
+    #[error("Invalid serialization")]
+    Serialization,
+
+    #[error("Invalid author: expected `{expected}`, got `None`")]
+    Validation1 { expected: DeviceID },
+
+    #[error("Invalid author: expected `{expected}`, got `{got}`")]
+    Validation2 { expected: DeviceID, got: DeviceID },
```
Comment (FirelightFlagboy) [inline oxidation/crates/parsec_api_types/src/error.rs:23]: `Validation2` is almost identical to `Validation1`.
Can those variant be merged together by using `Option` ?

Comment (TimeEngineer) [reply oxidation/crates/parsec_api_types/src/error.rs:23]: Good catch !

Comment (TimeEngineer) [reply oxidation/crates/parsec_api_types/src/error.rs:23]: The downside is adding Some around the id in the python error, which is a bit weird

Comment (TimeEngineer) [reply oxidation/crates/parsec_api_types/src/error.rs:23]: The change imply:

`DataError(f"Invalid author: expected `{repr_expected_author}`, got `{data.author}`")` 

---->

```Python
if data.author is None:
    `DataError(f"Invalid author: expected `{repr_expected_author}`, got `{data.author}`")` 
else:
    `DataError(f"Invalid author: expected `{repr_expected_author}`, got `Some({data.author})`")` 
```

Comment (TimeEngineer) [reply oxidation/crates/parsec_api_types/src/error.rs:23]: Another solution is to type `got` as `String`

Comment (TimeEngineer) [reply oxidation/crates/parsec_api_types/src/error.rs:23]: My bad, we can access self data

**Review by FirelightFlagboy — APPROVED** [review]

**Pull request merged**